### DISCUSSION TITLE
Callback on frame

### DISCUSSION THREAD

**Discussion by bss-tom-yaxley** [vizia/vizia#493] [Questions]
I'm building an application that requires me to update something every frame - is there a way of hooking in to the frame loop somehow? Maybe some event that gets fired at the start of each frame?

**Comment by geom3trik** [comment]
There's an `on_idle` callback on `Application` which allows you to hook into the event loop *after* all of the events have been handled. However, this will still only run when the event loop is woken up. You can wake up the event loop yourself by sending an event from another thread, or with a timer. Another option is to change the event loop control to poll rather than wait, which there's a function for on `Application` as well.

**Comment by bss-tom-yaxley** [comment]
I'll take a look at `on_idle`, it should suit my needs for now :)
In the latter option, I presume you're referring to [should_poll](https://docs.vizia.dev/vizia/struct.Application.html#method.should_poll)? Is there any documentation on what it does and how it should be used?

**Comment by geom3trik** [comment]
Still working on the documentation but essentially the should_poll function forces the event loop to wake up every frame, rather than just when an event is received.
